src/test/java/se/diabol/jenkins/pipeline/domain/PipelineTest.java

```diff
@@ -406,6 +413,16 @@ public void testPipelineLatestDownstreamIsDisabled() throws Exception {
 
     @Test
     public void testFirstUpstreamBuildFirstProjectHasJustOneUpstreamJob() throws Exception {
+        final Stapler stapler = Mockito.mock(Stapler.class);
+        final HttpServletRequest servletRequest = Mockito.mock(HttpServletRequest.class);
+
+        final StaplerRequest staplerRequest = new RequestImpl(stapler, servletRequest, Collections.EMPTY_LIST, null);
+        {
+            final WebApp webApp = Mockito.mock(WebApp.class);
+            Mockito.when(webApp.getClassLoader()).thenReturn(this.getClass().getClassLoader());
+            Mockito.when(stapler.getWebApp()).thenReturn(webApp);
+        }
```

Comment:
For getting "Page" number and "Component" number from Url, we are using StaplerRequest to get those details. For some of the tests required to Mock the StaplerRequest to avoid the exception and failure at time of getting CurrentStaplerRequest.